libs/sdk-ui-ext/src/insightView/InsightRenderer.tsx

```diff
@@ -247,17 +247,65 @@ class InsightRendererCore extends React.PureComponent<IInsightRendererProps & Wr
 
 export const IntlInsightRenderer = compose(injectIntl, withTheme, withContexts)(InsightRendererCore);
 
+/**
+ * Since InsightRendererImpl only sets callback (onDrill or pushData etc) on the first render (this is a PlugVis API, there is no way to update onDrill there),
+ * we have to update callback manually by this hook.
+ *
+ * This problem leads to unpredictable closures problems in using react hooks effect and dependencies arrays.
```

Comment:
If I understand it correctly, the issue was more about the fact, that the changed callbacks (with different reference) were not properly propagated to the "visualization" instance (as it took only the first provided callbacks).

Reply:
Yes you are right